spec/integ/crypto/rust-crypto.spec.ts

```diff
@@ -237,6 +247,143 @@ describe("MatrixClient.initRustCrypto", () => {
         // The final call should have progress == total == -1
         expect(progressListener).toHaveBeenLastCalledWith(-1, -1);
     }, 60000);
+
+    describe("Legacy trust migration", () => {
+        beforeEach(() => {
+            fetchMock.reset();
+        });
+
+        it("should not revert to untrusted if legacy was trusted but msk not in cache, big account", async () => {
+            // Just 404 here for the test
+            fetchMock.get("path:/_matrix/client/v3/room_keys/version", {
+                status: 404,
+                body: {
+                    errcode: "M_NOT_FOUND",
+                    error: "No backup found",
+                },
+            });
+
+            mockKeyQueryForTestMigration();
+
+            const testStoreName = "test-store";
+            await populateStore(testStoreName);
+            const cryptoStore = new IndexedDBCryptoStore(indexedDB, testStoreName);
+            await cryptoStore.startup();
```

Comment:
is it important that we use a different store name for each test? I don't think it should be - we reset indexeddb after each test.

I wonder if it would be good to factor out these 4 lines to a function so you can just do `const cryptoStore = await populateAndStartLegacyCryptoStore(dumpPath);`

Reply:
done